packages/jest-each/src/bind.js

```diff
@@ -85,16 +86,15 @@ const arrayFormat = (title, ...args) => {
       if (prettyIndexes.indexOf(index) !== -1) {
         return {
           args: acc.args,
-          title: acc.title.replace(
-            PRETTY_PLACEHOLDER,
-            pretty(arg, {maxDepth: 1, min: true}),
-          ),
+          title: acc.title
+            .replace(PRETTY_PLACEHOLDER, pretty(arg, {maxDepth: 1, min: true}))
+            .replace(INDEX_PLACEHOLDER, `${index}`),
```

Comment:
This index represents the index of each arg in a row of data. 

I think you want to use the index of each row in the outer table on line 32. 

```js
return table.forEach(row =>
  cb(arrayFormat(title, ...row), applyRestParams(row, test)),
);
```